src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object [] DEFAULT_EMPTY_ELEMENTDATA = {};
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialSize) {
+        if (initialSize > 0) {
+            elementData = new Object[initialSize];
+        } else if (initialSize == 0) {
+            elementData = DEFAULT_EMPTY_ELEMENTDATA;
+        } else {
+            throw new IllegalArgumentException("Illegal capacity");
+        }
+    }
+
+    public ArrayList(Collection<? extends T> c) {
+        Object [] listArray = c.toArray();
+        size = listArray.length;
+        if (size > 0) {
+            elementData = Arrays.copyOf(listArray, size);
+        } else {
+            elementData = DEFAULT_EMPTY_ELEMENTDATA;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeArrayIfNeeded();
+        elementData[size] = value;
```

Comment:
Minor thing, but you can(not necessarily) simplify this line to `elementData[size++] = value;` and don't write size++ on the next line.